File: netconnect.py
import subprocess

def test_port_connection(remote_server, *ports):
    for port in ports:
        print(f"Testing connectivity to {remote_server} on port {port}...")
        try:
            subprocess.run(["nc", "-z", "-w3", remote_server, str(port)], check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
            print(f"Connection to {remote_server} on port {port} succeeded.")
        except subprocess.CalledProcessError as e:
            print(f"Connection to {remote_server} on port {port} failed: {e.stderr.decode().strip()}")

def shutdown_ports(*ports):
    for port in ports:
        print(f"Shutting down port {port}...")
        # Stop-Process -Id (Get-NetTCPConnection -LocalPort $port).OwningProcess -Force
        # Uncomment the above line if you want to forcefully kill the process using the port

    print("All created ports have been shutdown.")

def show_menu(remote_server, *ports):
    print("========================")
    print("  NetConnect - Python Version")
    print("========================")
    print("1. Test Connectivity")
    print("2. Create Ports")
    print("3. Exit")
    print("========================")
    choice = input("Enter your choice: ")

    if choice == "1":
        test_port_connection(remote_server, *ports)
        show_menu(remote_server, *ports)
    elif choice == "2":
        ports_input = input("Enter the ports you want to create (space-separated): ")
        new_ports = [int(port) for port in ports_input.split()]
        for port in new_ports:
            print(f"Creating port {port}...")
            # Perform any specific actions needed for port creation
            # For demonstration purposes, we are not making any changes to the system.
        show_menu(remote_server, *(ports + new_ports))
    elif choice == "3":
        shutdown_ports(*ports)
        print("Exiting NetConnect...")
    else:
        print("Invalid choice. Please select a valid option.")
        show_menu(remote_server, *ports)

# Start the NetConnect tool
remote_server = input("Enter the remote server IP: ")
ports = []

show_menu(remote_server, *ports)

File: netconnect.pl
#!/usr/bin/perl

use strict;
use warnings;
use Net::Ping;

# Function to create TCP/UDP ports on the server
sub create_ports {
    my @ports = @_;
    foreach my $port (@ports) {
        system("nc -l -p $port &");
        system("nc -ul -p $port &");
    }
}

# Function to check connectivity with multiple servers via multiple ports
sub check_connectivity {
    my (@servers, @ports) = @_;

    foreach my $server (@servers) {
        foreach my $port (@ports) {
            print "Checking connectivity to $server on port $port...\n";
            my $p = Net::Ping->new();
            if ($p->ping($server)) {
                print "Ping successful.\n";
                my $result = `nc -zv $server $port 2>&1`;
                print "$result\n";
            } else {
                print "Ping failed.\n";
            }
        }
    }
}

# Main menu function
sub main_menu {
    print "==== TCP/UDP Connection Tool ====\n";
    print "1. Create TCP/UDP Ports\n";
    print "2. Check Connectivity with Other Servers\n";
    print "3. Exit\n";
    print "================================\n";

    my $choice = <STDIN>;
    chomp($choice);

    given ($choice) {
        when (1) {
            print "Enter the ports to create (separated by spaces, e.g., '8080 8888 9000'): ";
            my $ports_input = <STDIN>;
            chomp($ports_input);
            my @ports = split(/\s+/, $ports_input);
            create_ports(@ports);
        }
        when (2) {
            print "Enter the servers or IP addresses to check (separated by spaces, e.g., '192.168.1.100 192.168.1.200 10.0.0.1-10.0.0.10'): ";
            my $servers_input = <STDIN>;
            chomp($servers_input);
            my @servers = split(/\s+/, $servers_input);

            print "Enter the ports to check (separated by spaces, e.g., '80 443 8080'): ";
            my $ports_input = <STDIN>;
            chomp($ports_input);
            my @ports = split(/\s+/, $ports_input);

            check_connectivity(@servers, @ports);
        }
        when (3) {
            print "Exiting...\n";
            exit 0;
        }
        default {
            print "Invalid choice. Please try again.\n";
        }
    }

    main_menu();
}

# Start the script by displaying the main menu
main_menu();
